File: backend/backend/Areas/Identity/Models/UserRoles.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.AspNetCore.Identity;

namespace backend.Areas.Identity.Models;

public class UserRoles : IdentityUserRole<string>
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }
    public string UserId { get; set; }
    public string RoleId { get; set; }
    public virtual User User { get; set; }
    public virtual Role Role { get; set; }
}

File: frontend/lib/models/user_roles.dart
import 'package:frontend/models/User.dart';
import 'package:frontend/models/role.dart';

class UserRoles {
  int? id;
  String? userId;
  String? roleId;
  User? user;
  Role? role;

  UserRoles({
    this.id,
    this.userId,
    this.roleId,
    this.user,
    this.role,
  });

  factory UserRoles.fromJson(Map<String, dynamic> json) {
    return UserRoles(
      id: json['id'],
      userId: json['userId'],
      roleId: json['roleId'],
      user: json['user'] != null ? User.fromJson(json['user']) : null,
      role: json['role'] != null ? Role.fromJson(json['role']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'userId': userId,
      'roleId': roleId,
      'user': user?.toJson(),
      'role': role?.toJson(),
    };
  }
}